net/nimble/controller/src/ble_ll_scan.c

```diff
@@ -2386,13 +2398,12 @@ ble_ll_scan_rx_pkt_in(uint8_t ptype, struct os_mbuf *om, struct ble_mbuf_hdr *hd
             }
 
             ble_ll_scan_req_backoff(scansm, 1);
-            scansm->cur_aux_data = NULL;
         }
 
-        if (!BLE_LL_CHECK_AUX_FLAG(aux_data, BLE_LL_AUX_CHAIN_BIT)) {
-            /* If there is no chaining, we can remove data. Otherwise it is
-             * already scheduled for next aux*/
-            ble_ll_scan_aux_data_free(hdr->rxinfo.user_data);
+        if (BLE_LL_CHECK_AUX_FLAG(aux_data, BLE_LL_AUX_CHAIN_BIT)) {
+            // If there is chaining, we don't want to free aux data
```

Comment:
c++ style comments are not allowed

Reply:
actually this whole if is not needed. Chain support will come in different PR.